templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,43 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+csr = STDIN.read
+
+settings_file    = File.exist?('/etc/puppetlabs/puppet/foreman.yaml') ? '/etc/puppetlabs/puppet/foreman.yaml' : '/etc/puppet/foreman.yaml'
+SETTINGS         = YAML.load_file(settings_file)
+protocol         = "<%= scope.lookupvar("foreman_proxy::puppetca_listen_on") %>"
+port             = "<% if scope.lookupvar("foreman_proxy::puppetca_listen_on") == "https" %><%= scope.lookupvar("foreman_proxy::ssl_port") -%>
+                    <%- else %><%= scope.lookupvar("foreman_proxy::http_port") -%>
+                    <%- end %>"
+# e.g. https://hostname.localdomain.com:8443/puppet/ca/autosign
+uri              = URI.parse("#{protocol}://<%= @fqdn %>:#{port}/puppet/ca/autosign")
+res              = Net::HTTP.new(uri.host, uri.port)
+res.use_ssl      = true
+res.ca_file      = SETTINGS[:ssl_ca]
+res.verify_mode  = OpenSSL::SSL::VERIFY_PEER
+res.cert         = OpenSSL::X509::Certificate.new(File.read(SETTINGS[:ssl_cert]))
+res.key          = OpenSSL::PKey::RSA.new(File.read(SETTINGS[:ssl_key]), nil)
+res.open_timeout = SETTINGS[:timeout]
+res.read_timeout = SETTINGS[:timeout]
+req              = Net::HTTP::Post.new(uri.request_uri)
+req.add_field('Accept', 'application/json,version=2' )
+req.content_type = 'application/json'
+req.body         = {:csr => csr}.to_json
+begin
+    res.start do |http|
+        respone = http.request(req)
+        if response.code == 200
+            exit 0
```

Comment:
Nit: indentation is off.

Can also be written like this: `exit 0 if response.code == 200`

Do we need to suppport other response codes?